src/Filters/AuthRates.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace CodeIgniter\Shield\Filters;
+
+use CodeIgniter\Filters\FilterInterface;
+use CodeIgniter\HTTP\RedirectResponse;
+use CodeIgniter\HTTP\RequestInterface;
+use CodeIgniter\HTTP\Response;
+use CodeIgniter\HTTP\ResponseInterface;
+use CodeIgniter\Shield\Auth;
+
+/**
+ * Chain Authentication Filter.
+ *
+ * Checks all authentication systems specified within
+ * `Config\Auth->authenticationChain`
+ */
+class AuthRates implements FilterInterface
+{
+    /**
+     * Intened for use on auth form pages to restrict the number
+     * of attempts that can be generated. Restricts it to 10 attempts
+     * per minute, which is what auth0 uses.
+     *
+     * @see https://auth0.com/docs/troubleshoot/customer-support/operational-policies/rate-limit-policy/database-connections-rate-limits
```

Comment:
Auth0 does not matter.

Reply:
Do you mean the implementation of this filter referred to Auth0?

> If one IP address makes 20 login attempts in one minute to the same user account, the rate limit comes into effect. After that, Auth0 allows the user 10 attempts per minute. Any combination of successful and failed login attempts count toward this limit. 
https://auth0.com/docs/troubleshoot/customer-support/operational-policies/rate-limit-policy/database-connections-rate-limits

Reply:
Yes, that's why I linked the article. For us to see where I chose the rate limiting amount from. And to be able to go back and reference their other pages about how they protect against other attacks as inspiration. 

I understand that the company Auth0 doesn't matter for what we're building here, but as a large company with lots of data around the subject, they seemed like a good source of inspiration for how much to limit.

Reply:
Okay, I got it. That makes sense.